src/network.rs

```diff
@@ -229,52 +177,39 @@ impl Network {
         stats
     }
 
-    fn check_section_sizes(&self) -> bool {
-        if let Some(section) = self.sections.values().find(|section| {
-            section.nodes().len() > self.params.max_section_size
-        })
-        {
-            let prefixes = section.prefix().split();
-            let count0 =
-                node::count_matching_adults(&self.params, prefixes[0], section.nodes().values());
-            let count1 =
-                node::count_matching_adults(&self.params, prefixes[1], section.nodes().values());
-
-            error!(
-                "{}: {}: {} (adults per subsections: [..0]: {}, [..1]: {})",
-                log::prefix(&section.prefix()),
-                log::error("too many nodes"),
-                section.nodes().len(),
-                count0,
-                count1,
-            );
-            false
-        } else {
-            true
-        }
-    }
-}
-
-// Generate random `Live` request in the given section.
-fn add_random_node(params: &Params, section: &mut Section) {
-    let name = section.prefix().substituted_in(random::gen());
-    section.receive(Request::Live(Node::new(name, params.init_age)));
-}
+    fn validate(&self) {
+        for section in self.sections.values() {
+            if section.nodes().len() > self.params.max_section_size {
+                let prefixes = section.prefix().split();
+                let count0 = node::count_matching_adults(
+                    &self.params,
+                    prefixes[0],
+                    section.nodes().values(),
+                );
+                let count1 = node::count_matching_adults(
+                    &self.params,
+                    prefixes[1],
+                    section.nodes().values(),
+                );
+
+                error!(
+                    "{}: too many nodes: {} (adults per subsections: [..0]: {}, [..1]: {})",
+                    log::prefix(&section.prefix()),
+                    section.nodes().len(),
+                    count0,
+                    count1,
+                );
+            }
 
-// Generate random `Dead` request in the given section.
-fn drop_random_node(_params: &Params, section: &mut Section) -> bool {
-    let name = node::by_age(section.nodes().values())
-        .into_iter()
-        .find(|node| {
-            random::gen_bool_with_probability(node.drop_probability())
-        })
-        .map(|node| node.name());
-
-    if let Some(name) = name {
-        section.receive(Request::Dead(name));
-        true
-    } else {
-        false
+            let incoming = section.incoming_relocations();
+            if incoming.len() > 0 {
```

Comment:
why not just `if !section.incoming_relocations().is_empty() {}` ?
and, just wondering shall the outgoing_cache be checked as well?

Reply:
`incoming_relocations` returns `hash_set::Keys` which does not have `is_empty` (that is a nightly only feature). Also, I wanted to print the names of the nodes stuck in the cache, for easier debugging. Good point about `outgoing_relocations`, I'll add it.